File: LiftScout/app/src/main/java/com/brandonhogan/liftscout/views/about/AboutFragment.java
package com.brandonhogan.liftscout.views.about;

import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.brandonhogan.liftscout.BuildConfig;
import com.brandonhogan.liftscout.R;
import com.brandonhogan.liftscout.views.base.BaseFragment;

import butterknife.Bind;

public class AboutFragment extends BaseFragment {

    // Instance
    //
    public static AboutFragment newInstance() {
        return new AboutFragment();
    }


    // Private Properties
    //
    private View rootView;


    // Bindings
    //
    @Bind(R.id.version_number)
    TextView versionNumber;


    //Overrides
    //
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        rootView = inflater.inflate(R.layout.frag_about, null);
        return rootView;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        setTitle(getResources().getString(R.string.title_frag_about));


        versionNumber.setText(BuildConfig.VERSION_NAME);
    }
}

File: app/src/main/java/com/brandonhogan/liftscout/features/about/AboutFragment.kt
package com.brandonhogan.liftscout.features.about

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import androidx.lifecycle.lifecycleScope
import com.brandonhogan.liftscout.R
import com.brandonhogan.liftscout.injection.Injector
import com.brandonhogan.liftscout.repository.objects.About
import com.brandonhogan.liftscout.views.base.BaseFragment
import kotlinx.android.synthetic.main.about_fragment.*
import kotlinx.coroutines.*

class AboutFragment : BaseFragment() {

    companion object {
        @JvmStatic
        fun newInstance() = AboutFragment()
    }

    //private val viewModel: AboutViewModel by viewModels {  Injector.provideAboutViewModelFactory() }
    private lateinit var viewModel: AboutViewModel

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.about_fragment, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        lifecycleScope.launch {

            viewModel = ViewModelProviders.of(
                    this@AboutFragment,
                    Injector.provideAboutViewModelFactory()
            ).get(AboutViewModel::class.java)


            withContext(Dispatchers.Main) {
                viewModel.getAbout().observe(viewLifecycleOwner, Observer { about ->
                    testName.text = about?.name
                })
            }
        }

        // When button is clicked, instantiate a Comment and add it to DB through the ViewModel
        testButton.setOnClickListener {
            GlobalScope.launch {
                val about = About(
                    "Does this still work while within a launch/async",
                    "2.2.2"
                )
                viewModel.setAbout(about)
            }
        }
    }
}